packages/webapp-libs/webapp-tenants/src/providers/currentTenantProvider/currentTenantProvider.utils.ts

```diff
@@ -0,0 +1,49 @@
+import { CommonQueryTenantItemFragmentFragment } from '@sb/webapp-api-client';
+import { TenantType } from '@sb/webapp-api-client/constants';
+import { prop } from 'ramda';
+
+export const parseStoredState = (storedState: string, userId = '') => {
+  try {
+    const parsedStoredState = JSON.parse(storedState);
+
+    return {
+      parsedStoredState,
+      storedTenantId: parsedStoredState?.[userId] ?? null,
+    };
+  } catch {
+    return {
+      parsedStoredState: {},
+      storedTenantId: null,
+    };
+  }
+};
+
+const getTenantId = (
+  tenantId = '',
+  storedTenantId: string | null,
+  tenants: (CommonQueryTenantItemFragmentFragment | null | undefined)[]
+) => {
+  if (tenantId) return tenantId;
+
+  if (storedTenantId && tenants.map(prop<string>('id')).includes(storedTenantId)) {
+    return storedTenantId;
+  }
+
+  return '';
+};
+
+export const getCurrentTenant = (
+  paramsTenantId = '',
+  storedTenantId: string | null,
+  tenants: (CommonQueryTenantItemFragmentFragment | null | undefined)[]
+) => {
+  const tenantId = getTenantId(paramsTenantId, storedTenantId, tenants);
+
+  const currentTenant = tenants.find((t) => t?.id === tenantId && t.membership.invitationAccepted);
+  if (currentTenant) return currentTenant;
+
+  const firstDefaultTenant = tenants.find((t) => t?.type === TenantType.PERSONAL);
+  if (firstDefaultTenant) return firstDefaultTenant;
```

Comment:
I was thinking... if there are no `currentTenant`, we could set `callback` to navigate to the proper ID.
Right now we are just setting the currentTenant without any change of URL.

example:
```
if(!tenantId) callback(firstDefaultTenant?.id ?? tenants[0]?.id)
if (firstDefaultTenant) return firstDefaultTenant;
```

Reply:
For now lets keep it as it is